### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by vr4b4c**
### Aim & Solution
Update README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,11 +31,70 @@ module Api
...
 
-spec/support/infinum_json_api_setup.rb
+`respond_with` is well integrated with `ActiveRecord::Model` interface. Given a compliant object, the method will correctly set a response status and handle object(or error) serialization based on the presence of `.errors`. For a successful domain operation, HTTP status will be 200 OK(or 204 in case of `create` controller action). Unsuccessful operations will have HTTP status 422 Unprocessable Entity with errors structured according to [JSON:API specification](https://jsonapi.org/format/#error-objects).
+```ruby
+def create
+  respond_with User.create(permitted_params)
+end
+```
 
+`respond_with` also detects usage from a `destroy` controller action and responds with HTTP status 200 OK and an empty body.
```
Comment (nikajukic) [inline README.md:61]: Is it maybe 204 No Content?

File: README.md [hunk 2]
```diff
@@ -27,11 +31,70 @@ module Api
...
+`InfinumJsonApiSetup::JsonApi::ContentNegotiation` module is designed to integrate [server responsibilities](https://jsonapi.org/format/#content-negotiation-servers) of content negotiation protocol described by the JSON:API specification.
+
+### Error handling
+`InfinumJsonApiSetup::JsonApi::ErrorHandling` module is designed to catch and handle common exceptions that might bubble up when processing a request.
+
+| Exception                                      | HTTP status | Bugsnag notification |
+| ---                                            | :---:       | :---:                |
+| `ActionController::ParameterMissing`           | 400         |                      |
+| `ActionDispatch::Http::Parameters::ParseError` | 400         |                      |
+| `Jure::UnpermittedSortParameters`              | 400         | :white_check_mark:   |
```
Comment (nikajukic) [inline README.md:81]: Jure? 😂

**Review by uncoverd — APPROVED** [review]
🎉

**Review by nikajukic — COMMENTED** [review]
Should we also document request helpers and show the example of the body you get.

**Comment by vr4b4c** [comment]
Definitely, I just don't think now is the right moment to try and cover everything. I have an extensive TODO list with ideas and enhancements but had to draw a line somewhere...

File: README.md (hunk 2 above)
Comment (vr4b4c) [reply README.md:81]: Yep, his deep nesting negatively impacts my table layout, so I decided to plant an easter egg. Kind of like it tbh...

File: README.md (hunk 1 above)
Comment (vr4b4c) [reply README.md:61]: That's what you get when you don't fact-check [SO answer](https://stackoverflow.com/a/15681226). Absolutely correct

**Review by djordjelacmanovic — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -27,11 +31,70 @@ module Api
...
+### Responding
+Use `respond_with` to initiate transformation(serialization) of domain objects into HTTP response.
+```ruby
+def show
+  respond_with User.find(params[:id])
+end
+```
 
-spec/support/infinum_json_api_setup.rb
+`respond_with` is well integrated with `ActiveRecord::Model` interface. Given a compliant object, the method will correctly set a response status and handle object(or error) serialization based on the presence of `.errors`. For a successful domain operation, HTTP status will be 200 OK(or 204 in case of `create` controller action). Unsuccessful operations will have HTTP status 422 Unprocessable Entity with errors structured according to [JSON:API specification](https://jsonapi.org/format/#error-objects).
```
[the remark this answers is not in the document]
Comment (vr4b4c) [reply README.md]: taken into account, I did an update with consistent whitespace&parenthesis across the whole document. But didn't commit your exact suggestion as there were other stuff I changed on the same line

**Pull request merged**